src/main/java/org/opensearch/testcontainers/OpensearchContainer.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.opensearch.testcontainers;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+import static java.net.HttpURLConnection.HTTP_UNAUTHORIZED;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.utility.Base58;
+import org.testcontainers.utility.DockerImageName;
+
+/**
+ * The Opensearch Docker container (single node cluster) which exposes by default ports 9200
+ * (http/https) and 9300 (tcp, deprecated).
+ */
+public class OpensearchContainer extends GenericContainer<OpensearchContainer> {
+    // Default username to connect to Opensearch instance
+    private static final String DEFAULT_USER = "admin";
+    // Default password to connect to Opensearch instance
+    private static final String DEFAULT_PASSWORD = "admin";
+
+    // Default HTTP port.
+    private static final int DEFAULT_HTTP_PORT = 9200;
+
+    // Default TCP port (deprecated and may be removed in future versions).
+    @Deprecated
+    private static final int DEFAULT_TCP_PORT = 9300;
```

Comment:
Do we really need this? You can save yourself some future work by not introducing the deprecating thing in the first place :)

Reply:
We sadly need this because we support 1.3.x (with transport client) and 2.x (without transport client).

Reply:
I think it would be fine to only add the `@Deprecated` annotation to the `getTcpHost()` method.